### PR TITLE
setUp-PullRequest

### PR DESCRIPTION
**Pull request by IvonneBenitesRodriguez**
**About this project** <br/>

This app has the functionality to show greetings in several languages, the app is made in Rails and React.

**For it :**<br/>

A table for storing the messages and create 5 different greetings was created.
An API endpoint that selects a random greeting from the table was created.
A static view that will be the root of the app was created.
A Greeting component that will display a greeting is created and it is  set up as a route in the App component.
A store, an action, and a reducer that will connect to the API endpoint to get the random greeting is created.
A  display the random greeting in the Greeting component is created.
Linters were passed too.

**Goal:**<br/>
The goal is to understand the usage of rails and react together, also how to connect all the structure with the database.

### CODE REVIEW

File: db/migrate/20240205171918_create_greetings.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class CreateGreetings < ActiveRecord::Migration[7.0]
+  def change
+    create_table :greetings do |t|
+
+      t.timestamps
+    end
```
Comment (thecodechaser) [inline db/migrate/20240205171918_create_greetings.rb]: - [x] There's no column in the `greeting` table to save the greetings in the database. Please make sure to add a column to the table. (the column name should be same as used in the API controller).

File: app/controllers/api/greetings_controller.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class Api::GreetingsController < ApplicationController
+  def random_greeting
+    random_message = Message.order('RANDOM()').first
+    render json: { greeting: random_message&.message }
+  end
```
Comment (thecodechaser) [inline app/controllers/api/greetings_controller.rb]: - [x] In the database schema file the table name is `greeting` and here you're trying to fetch greetings from the `message` table. Please make sure to fix this issue.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @IvonneBenitesRodriguez👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Good job with Hello World project. ✔️ 
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20240205171918_create_greetings.rb (hunk 1 above)
Comment (IvonneBenitesRodriguez) [reply db/migrate/20240205171918_create_greetings.rb]: Dear @thecodechaser, 
Thanks for the suggestion, I appreciate it, 
I solved it .
Best wishes,

File: app/controllers/api/greetings_controller.rb (hunk 2 above)
Comment (IvonneBenitesRodriguez) [reply app/controllers/api/greetings_controller.rb]: Dear @thecodechaser 
Thanks for the suggestion, I appreciate it, 
I solved it .
Best wishes,

**Comment by prolajumokeoni** [comment]
Hi @IvonneBenitesRodriguez, your app does not work when I  run `./bin/dev`, please this is a comment not a review 👍🏿 also in your `application.rb` file `    config.autoload_paths << Rails.root.join("lib")` which is causing an error

File: db/migrate/20240205171918_create_greetings.rb [hunk 3]
```diff
@@ -1,8 +1,9 @@
...
+#       t.string :message
 
-      t.timestamps
-    end
-  end
-end
+#       t.timestamps
+#     end
+#   end
+# end
```
Comment (Strangeal) [inline db/migrate/20240205171918_create_greetings.rb]: - [x] Fantastic work on addressing the recommendations from the previous review! 👏 Your responsiveness and dedication to refining your code are truly commendable.

   - However, I noticed a small hiccup. It appears that your app is encountering an error during runtime. The issue stems from the fact that you've commented out your table. As a result, when running the database migrations, errors are thrown.

  - Let's swiftly resolve this together! Un-commenting the table and ensuring it aligns with your database schema should do the trick. If you encounter any roadblocks. See below for more details on the error:
  
Screenshot 2024-02-05 at 22 32 20

Happy coding, and keep up the excellent work! ✅👨‍💻

**Review by Strangeal — CHANGES_REQUESTED** [review]
Hi @IvonneBenitesRodriguez, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

Alt Text

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20240205171918_create_greetings.rb (hunk 3 above)
Comment (IvonneBenitesRodriguez) [reply db/migrate/20240205171918_create_greetings.rb]: Dear @Strangeal 🌷
Very Nice to meet you,  
I hope this message finds you well. 
Thanks for the suggestions.
The changes are done, and the app is working,
please approve my work to continue in the other projects 🙏🏿
Have a good Tuesday,
Best regards,
Ivonne

Comment (IvonneBenitesRodriguez) [reply db/migrate/20240205171918_create_greetings.rb]: Besides, I have checked and this is my last option for a review,
Please consider it 🙏🏿
Thank you so much for your kindness and consideration, 🙏🏿
Best wishes,

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @IvonneBenitesRodriguez,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**